internal/controller/dashboard_controller.go

```diff
@@ -0,0 +1,344 @@
+			return ctrl.Result{}, errors.WithStack(err)
+		}
+		logger.Info("added finalizer", "finalizer", DashboardFinalizer)
+		return ctrl.Result{}, nil
+	}
+
+	// Configure the dashboard in Grafana
+	if err := r.configureDashboard(ctx, dashboard); err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func getDashboardUID(jsonDashboard string) (UID string, err error) {
+
+	var Dashboard map[string]interface{}
+	err = json.Unmarshal([]byte(jsonDashboard), &Dashboard)
+	if err != nil {
+		return "", err
+	}
+
+	UID, ok := Dashboard["uid"].(string)
+	if !ok {
+		return "", errors.New("dashboard UID not found in configmap")
+	}
+	return UID, nil
+}
+
+func getDashboardCMOrg(dashboard *v1.ConfigMap) (Org string, err error) {
+
+	// Try to look for an annotation first
+	orgName := dashboard.GetAnnotations()[DashboardTeamAnnotation]
+	if orgName != "" {
+		return orgName, nil
+	}
+
+	// Then look for a label
+	orgName = dashboard.GetLabels()[DashboardTeamAnnotation]
+
+	if orgName != "" {
+		return orgName, nil
+	}
+
+	// Return an error if no label was found
+	return "", errors.New("No organization label found in configmap")
+}
+
+func (r DashboardReconciler) configureDashboard(ctx context.Context, dashboardCM *v1.ConfigMap) (err error) {
+	logger := log.FromContext(ctx)
+
+	dashboardOrg, err := getDashboardCMOrg(dashboardCM)
+	if err != nil {
+		logger.Info("Skipping dashboard, no organization found")
+		return nil
+	}
+
+	// We always switch back to the shared org
+	defer func() {
+		if _, err = r.GrafanaAPI.SignedInUser.UserSetUsingOrg(grafana.SharedOrg.ID); err != nil {
+			logger.Error(err, "failed to change current org for signed in user")
+		}
+	}()
+
+	for _, dashboard := range dashboardCM.Data {
+		dashboardUID, err := getDashboardUID(dashboard)
+		if err != nil {
+			logger.Info("Skipping dashboard, no UID found")
+			continue
+		}
+		// Switch context to the current org
+		organization, err := grafana.FindOrgByName(r.GrafanaAPI, dashboardOrg)
+		if err != nil {
+			logger.Error(err, "failed to find organization", "organization", dashboardOrg)
+			return errors.WithStack(err)
+		}
+		logger.Info("Found dashboard org", "Organization", organization)
+
+		if _, err = r.GrafanaAPI.SignedInUser.UserSetUsingOrg(organization.ID); err != nil {
+			logger.Error(err, "failed to change current org for signed in user")
+			return errors.WithStack(err)
+		}
+
+		var dashboardjson interface{}
+		err = json.Unmarshal([]byte(dashboard), &dashboardjson)
+		if err != nil {
+			logger.Info("Failed converting dashboard to json")
+			return errors.WithStack(err)
+		}
+
+		// Create or update dashboard
+		_, err = r.GrafanaAPI.Dashboards.PostDashboard(&models.SaveDashboardCommand{
+			UpdatedAt: strfmt.DateTime(time.Now()),
+			Dashboard: dashboardjson,
+			FolderID:  0,
+			FolderUID: "",
+			IsFolder:  false,
+			Message:   "Added by observability-operator",
+			Overwrite: true,
```

Comment:
What's the dashboard identifier used to overwrite it ? Is it by name ? Please document this.